watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1173,8 +1208,65 @@ def rule_visc_k_phase(b, p):
     def _radius_stokes_comp(self):
         add_object_reference(self, "radius_stokes_comp", self.params.radius_stokes_comp)
 
+    def _molar_volume_comp(self):
+        add_object_reference(self, "molar_volume_comp", self.params.molar_volume_comp)
+
     def _diffus_phase_comp(self):
-        add_object_reference(self, "diffus_phase_comp", self.params.diffus_phase_comp)
+        self.diffus_phase_comp = Var(
+            self.params.phase_list,
+            self.params.ion_set | self.params.solute_set,
+            initialize=1e-9,
+            units=pyunits.m**2 * pyunits.s**-1,
+            doc="Bulk diffusivity of ion or molecule",
+        )
+
+        def rule_diffus_phase_comp(b, p, j):
+            if self.params.config.diffus_calculation == DiffusivityCalculation.none:
+                if (p, j) not in self.params.config.diffusivity_data.keys():
+                    raise ConfigurationError(
+                        """ 
+                        Missing the "diffusivity_data" configuration to build the diffus_phase_comp 
+                        and/or its derived variables for {} in {}. 
+                        Provide this configuration or use DiffusivityCalculation.hayduklaudie.
+                        """.format(
+                            j, self.name
+                        )
+                    )
+                else:
+                    return (
+                        b.diffus_phase_comp[p, j]
+                        == self.params.config.diffusivity_data[p, j]
+                        * pyunits.m**2
+                        * pyunits.s**-1
+                    )
```

Comment:
What if we just keep the `add_object_reference` line here for the case when `DiffusivityCalculation` is none?